File: app/src/main/java/com/raival/fileexplorer/tab/file/FileExplorerTabDataHolder.java
package com.raival.fileexplorer.tab.file;

import android.os.Parcelable;

import com.raival.fileexplorer.tab.BaseDataHolder;
import com.raival.fileexplorer.tab.file.model.FileItem;

import java.io.File;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

public class FileExplorerTabDataHolder extends BaseDataHolder {
    private final String tag;
    public File activeDirectory;
    public Map<File, Parcelable> recyclerViewStates = new HashMap<>();
    public ArrayList<FileItem> searchList = new ArrayList<>();
    public ArrayList<File> selectedFiles = new ArrayList<>();

    public FileExplorerTabDataHolder(String tag) {
        this.tag = tag;
    }

    @Override
    public String getTag() {
        return tag;
    }
}

File: app/src/main/java/com/raival/fileexplorer/tab/file/FileExplorerTabDataHolder.kt
package com.raival.fileexplorer.tab.file

import android.os.Parcelable
import com.raival.fileexplorer.tab.BaseDataHolder
import com.raival.fileexplorer.tab.file.model.FileItem
import java.io.File

class FileExplorerTabDataHolder(override val tag: String) : BaseDataHolder() {
    @JvmField
    var activeDirectory: File? = null

    @JvmField
    var recyclerViewStates: HashMap<File, Parcelable> = HashMap()
    var searchList = ArrayList<FileItem>()

    @JvmField
    var selectedFiles = ArrayList<File>()
}